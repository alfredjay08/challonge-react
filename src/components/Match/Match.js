import React from "react";
import { MatchContainer } from "../StyledComponents/StyledComponents";
import Player from "../Player";

const Match = ({ matchId, players, isHighlighted, hovered, unhovered }) => {
  const activePlayers = players && players.length ? players : [null, null];

  return (
    <MatchContainer data-id={matchId}>
      {activePlayers.map((player, index) => (
        <Player
          key={index}
          highlighted={
            isHighlighted.status && player?.data._id !== isHighlighted.playerId
              ? "true"
              : "false"
          }
          hovered={players && hovered}
          unhovered={unhovered}
          playerInfo={players && player?.data}
          winner={players && player?.winner ? 1 : 0}
          playerScore={players && player?.score}
        />
      ))}
    </MatchContainer>
  );
};

export default Match;
