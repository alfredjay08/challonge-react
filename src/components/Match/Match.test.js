import React from "react";
import renderer from "react-test-renderer";

import Match from "./Match";

it("renders when there are no props", () => {
  const tree = renderer.create(<Match />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders when passed wrong properties", () => {
  const tree = renderer
    .create(<Match wrongProp={4} wrongStyle={{ background: "red" }} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders when there are one player", () => {
  const players = [
    {
      data: {
        _id: 1001,
        name: "Alfred Jay",
      },
      winner: true,
      score: 5,
    },
  ];
  const tree = renderer.create(<Match players={players} />).toJSON();

  expect(tree).toMatchSnapshot();
});
