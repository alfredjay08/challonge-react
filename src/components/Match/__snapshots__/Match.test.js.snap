// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when passed wrong properties 1`] = `
<div
  className="sc-eDvSVe LqqaX"
>
  <div
    className="sc-jSUZER hFnVAg"
    onMouseOver={[Function]}
  >
    <div
      className="sc-gKPRtg jAzlAk"
    >
      <img
        alt="Player Icon"
        src="/uploads/ninja-yellow.jpg"
        style={
          Object {
            "borderRadius": "50%",
            "width": "20px",
          }
        }
      />
    </div>
    <div
      className="sc-gKPRtg fOrJSf"
    />
    <div
      className="sc-gKPRtg jLsyYQ"
    >
      <span />
    </div>
  </div>
  <div
    className="sc-jSUZER hFnVAg"
    onMouseOver={[Function]}
  >
    <div
      className="sc-gKPRtg jAzlAk"
    >
      <img
        alt="Player Icon"
        src="/uploads/ninja-yellow.jpg"
        style={
          Object {
            "borderRadius": "50%",
            "width": "20px",
          }
        }
      />
    </div>
    <div
      className="sc-gKPRtg fOrJSf"
    />
    <div
      className="sc-gKPRtg jLsyYQ"
    >
      <span />
    </div>
  </div>
</div>
`;

exports[`renders when there are no props 1`] = `
<div
  className="sc-eDvSVe LqqaX"
>
  <div
    className="sc-jSUZER hFnVAg"
    onMouseOver={[Function]}
  >
    <div
      className="sc-gKPRtg jAzlAk"
    >
      <img
        alt="Player Icon"
        src="/uploads/ninja-yellow.jpg"
        style={
          Object {
            "borderRadius": "50%",
            "width": "20px",
          }
        }
      />
    </div>
    <div
      className="sc-gKPRtg fOrJSf"
    />
    <div
      className="sc-gKPRtg jLsyYQ"
    >
      <span />
    </div>
  </div>
  <div
    className="sc-jSUZER hFnVAg"
    onMouseOver={[Function]}
  >
    <div
      className="sc-gKPRtg jAzlAk"
    >
      <img
        alt="Player Icon"
        src="/uploads/ninja-yellow.jpg"
        style={
          Object {
            "borderRadius": "50%",
            "width": "20px",
          }
        }
      />
    </div>
    <div
      className="sc-gKPRtg fOrJSf"
    />
    <div
      className="sc-gKPRtg jLsyYQ"
    >
      <span />
    </div>
  </div>
</div>
`;

exports[`renders when there are one player 1`] = `
<div
  className="sc-eDvSVe LqqaX"
>
  <div
    className="sc-jSUZER hFnVAg"
    onMouseOver={[Function]}
  >
    <div
      className="sc-gKPRtg jAzlAk"
    >
      <img
        alt="Player Icon"
        src="/uploads/ninja-yellow.jpg"
        style={
          Object {
            "borderRadius": "50%",
            "width": "20px",
          }
        }
      />
    </div>
    <div
      className="sc-gKPRtg fOrJSf"
    >
      Alfred Jay
    </div>
    <div
      className="sc-gKPRtg zysMT"
    >
      <span>
        5
      </span>
    </div>
  </div>
</div>
`;
